database/storage_engine.c

```diff
@@ -83,7 +83,8 @@ static STORAGE_ENGINE engines[] = {
         }
     },
 #endif
-    { .id = RRD_MEMORY_MODE_NONE, .name = NULL }
+    // End-of-list
```

Comment:
Hello @aberaud ,

I remember this was inserted with this [PR](https://github.com/netdata/netdata/pull/12808), and I understand your goal with this PR, but I will give a suggestion. Considering that when compiler allocates this array the address for `id` will be allocated, but it will not be set, I suggest we keep this value and instead to remove `.id`, we could set `.api = NULL`. What do you think? This will force the compiler to set values for all addresses we have.

Reply:
I updated the patch to add RRD_MEMORY_MODE_INVALID at the end of the enum.

Reply:
Perfect, thanks @aberaud !